cli/cli.go

```diff
@@ -0,0 +1,180 @@
+package cli
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/exercism/cli/debug"
+	update "github.com/inconshreveable/go-update"
+)
+
+var (
+	// BuildOS is the operating system (GOOS) used during the build process.
+	BuildOS string
+	// BuildARM is the ARM version (GOARM) used during the build process.
+	BuildARM string
+	// BuildARCH is the architecture (GOARCH) used during the build process.
+	BuildARCH string
+)
+
+var (
+	osMap = map[string]string{
+		"darwin":  "mac",
+		"linux":   "linux",
+		"windows": "windows",
+	}
+
+	archMap = map[string]string{
+		"amd64": "64bit",
+		"386":   "32bit",
+		"arm":   "arm",
+	}
+)
+
+var (
+	// HTTPClient is the client used to make HTTP calls in the cli package.
+	HTTPClient = &http.Client{Timeout: 10 * time.Second}
+	// LatestReleaseURL is the endpoint that provides information about the latest release.
+	LatestReleaseURL = "https://api.github.com/repos/exercism/cli/releases/latest"
+)
+
+// CLI is information about the CLI itself.
+type CLI struct {
+	Version       string
+	LatestRelease *Release
+}
+
+// New creates a CLI, setting it to a particular version.
+func New(version string) *CLI {
+	return &CLI{
+		Version: version,
+	}
+}
+
+// IsUpgradeNeeded compares the current version to that of the latest release.
+func (c *CLI) IsUpgradeNeeded() (bool, error) {
```

Comment:
That'd make sense, I agree the name could be more idiomatic and using `ok` feels a bit better. `IsUpToDate` would be my choice if you asked me 😄